File: 0268-missing-number/0268-missing-number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        nums_dict = dict()
        
        for num in nums:
            nums_dict[num] = num
        
        for num in range(len(nums)):
            if num not in nums_dict:
                return num
        
        return len(nums)
        

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int size = nums.length;
        int[] arr = new int[size+1];
        
        for(int num: nums){
            arr[num]++;
        }
        
        for(int i = 0; i < size+1 ; i++){
            if(arr[i]==0){
                return i;
            }
        }
        
        return size;
    }
}